File: backend/src/main/java/com/deliverymatcher/backend/model/Application.java
package com.deliverymatcher.backend.model;


import jakarta.persistence.*;

import java.time.LocalDate;
import java.util.List;

@Entity
@Table(name = "applications")
public class Application {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "application_date")
    private LocalDate applicationDate;

    @Enumerated(EnumType.STRING)
    @Column(name = "application_status")
    private ApplicationStatus applicationStatus;

    @ManyToOne
    private Announcement announcement;

    @ManyToOne
    private Sender sender;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "package_id", referencedColumnName = "id")
    private Package pack;

}

File: frontend/src/app/models/interfaces/application.ts
import {Announcement} from './announcement';
import {Sender} from './sender';
import {Pack} from './pack';

export interface Application {
  id: number | null,
  applicationDate: Date | null,
  applicationStatus: string | null,
  announcement: Announcement | null,
  sender: Sender | null,
  pack: Pack | null
}
